### PR TITLE
[MM-44123] Advanced Text Editor Feature Flag

### PR DESCRIPTION
**Pull request by AshishDhama**
#### Summary
Add Advanced Text Editor Feature Flag

#### Ticket Link
  Fixes https://mattermost.atlassian.net/browse/MM-44123
  Epic: https://mattermost.atlassian.net/browse/MM-41347

#### Release Note
```release-note
* Add Advanced Text Editor Feature Flag
```

### CODE REVIEW

**Review by agnivade — APPROVED** [review]

**Review by its0x08 — APPROVED** [review]

File: model/feature_flags.go [hunk 1]
```diff
@@ -70,7 +70,8 @@ type FeatureFlags struct {
 
 	CloudFree bool
 
-	CommandPalette bool
+	CommandPalette    bool
+	AdvanceTextEditor bool
```
Comment (Willyfrog) [inline model/feature_flags.go:74]: There is a typo, no big deal but usually this things cause confusion down the line.
```suggestion
	AdvancedTextEditor bool
```

File: model/feature_flags.go [hunk 2]
```diff
@@ -96,6 +97,7 @@ func (f *FeatureFlags) SetDefaults() {
 	f.InsightsEnabled = false
 	f.CloudFree = false
 	f.CommandPalette = false
+	f.AdvanceTextEditor = false
```
Comment (Willyfrog) [inline model/feature_flags.go:100]:
```suggestion
	f.AdvancedTextEditor = false
```

**Review by Willyfrog — APPROVED** [review]
non blocking comment

**Review by vish9812 — APPROVED** [review]

**Pull request merged**